stride/utils/fullwave.py

```diff
@@ -0,0 +1,329 @@
+
+import numpy as np
+import struct
+import mosaic
+
+
+__all__ = ['read_vtr_model3D', 'read_observed_ttr', 'read_signature_ttr',
+            'read_signature_txt', 'read_geometry_pgy']
+
+
+def read_vtr_model3D(vtr_path, swapaxes=False):
+    '''
+    Function to read 3D vtr low kernel binary data (model files) and
+    returns the data as an as ndarray.
+
+    Code adapted from Oscar Calderon from Imperial College London's
+    FULLWAVE consortium.
+
+    Parameters
+    ----------
+    vtr_path : str
+        The path to the vtr file, which contains the 3D model
+    swapaxes : bool, optional
+        Permutes Fullwave storing format (depth, cross-line, in-line) to stride format (in-line,
+        depth, cross-line). Default False.
+
+    Returns
+    -------
+    model : ndarray
+        File that was read from vtr
+    '''
+    with open(vtr_path, 'rb') as f:
+        # Read headers from binary file.
+        rec_len = np.fromfile(f, dtype='int32', count=1)
+        ncomp = np.fromfile(f, dtype='int32', count=1)
+        ndim = np.fromfile(f, dtype='int32', count=1)
+        extranum = np.fromfile(f, dtype='int32', count=1)
+
+        rec_len = np.fromfile(f, dtype='int32', count=1)
+        rec_len = np.fromfile(f, dtype='int32', count=1)
+        n3 = np.fromfile(f, dtype='int32', count=1)
+        n2 = np.fromfile(f, dtype='int32', count=1)
+        n1 = np.fromfile(f, dtype='int32', count=1)
+        rec_len = np.fromfile(f, dtype='int32', count=1)
+
+        # Read model:
+        ind1 = 0
+        ind2 = -1
+        model = np.zeros((int(n3), int(n2), int(n1)))
+        trace = np.zeros((int(n3), int(1)))
+        while True:
+            ind2 = ind2 + 1
+            if ind2 > n2-1:
+                ind2 = 0
+                ind1 = ind1 + 1
+                if ind1 > n1-1:
+                    break
+
+            rec_len = np.fromfile(f, dtype='int32', count=1)
+            if rec_len.size < 1:
+                break
+            else:
+                trace = np.fromfile(f, dtype='f4', count=int(n3))
+                rec_len = np.fromfile(f, dtype='int32', count=1)
+                model[:, ind2, ind1] = trace[:]
+    if swapaxes:
+        model = np.swapaxes(np.swapaxes(model, 0, 1), 0, 2)
+    return model
+
+
+def read_header_ttr(ttr_path):
+    """
+    Function to read ttr header.
+    Parameters
+    ----------
+    ttr_path : str
+        The path to the ttr file, which contains metadata about model geomtery.
+
+    Returns
+    -------
+    number_composite_shots : int
```

Comment:
for consistency, `num_composite_shots`

Reply:
Amended, with dependent change on line 103, 104